src/libraries/System.Text.Json/tests/Serialization/ContinuationTests.cs

```diff
@@ -0,0 +1,511 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class ContinuationTests
```

Comment:
The first four tests in this file are very similar, the only difference being the inner type passed to outer, and also an `Assert.Null` test if the innermost type is a reference type. To avoid the duplication: can we consolidate them into one generic test helper, and run the test like this?

```cs
[Theory]
[MemberData(nameof(TestData))]
public static async Task ContinuationShouldWorkAtAnyPosition_ParameterlessCtor(int paddingLength, bool ignoreNullValues)
{
    await RunTest<TestClass<NestedClass>, NestedClass>(paddingLength, ignoreNullValues);
    await RunTest<TestClass<NestedValueType>, NestedValueType>(paddingLength, ignoreNullValues);
    await RunTest<TestValueType<NestedClass>, NestedClass>(paddingLength, ignoreNullValues);
    await RunTest<TestValueType<NestedValueType>, NestedValueType>(paddingLength, ignoreNullValues);
}

private static async Task RunTest<TInner, TInnermost>(int paddingLength, bool ignoreNullValues)
    where TInner : ITestInterface<TInnermost>, new()
    where TInnermost : INestedInterface, new()
{
    var stream = new MemoryStream();
    {
        var obj = new Outer<TInner>
        {
            S = new string('x', paddingLength),
            C = new()
            {
                A = "Hello",
                B = null,
                C = 42,
                D = null,
                E = 3.14e+17f,
                F = null,
                G = true,
                H = null,
                I = new int[] { 42, 17 },
                J = new()
                {
                    A = null,
                    B = 7,
                }
            }
        };
        await JsonSerializer.SerializeAsync(stream, obj, new JsonSerializerOptions { Converters = { new OuterConverter<TestClass<NestedClass>>() } });
    }

    stream.Position = 0;
    {
        var readOptions = new JsonSerializerOptions
        {
            DefaultBufferSize = 128,
            IgnoreNullValues = ignoreNullValues,
        };

        Outer<TestClass<NestedClass>> obj = await JsonSerializer.DeserializeAsync<Outer<TestClass<NestedClass>>>(stream, readOptions);

        Assert.Equal(new string('x', paddingLength), obj.S);
        Assert.Equal("Hello", obj.C.A);
        Assert.Null(obj.C.B);
        Assert.Equal(42, obj.C.C);
        Assert.Null(obj.C.D);
        Assert.Equal(3.14e17f, obj.C.E);
        Assert.Null(obj.C.F);
        Assert.True(obj.C.G);
        Assert.Null(obj.C.H);
        Assert.Collection(obj.C.I, v => Assert.Equal(42, v), v => Assert.Equal(17, v));

        bool innerIsRefType = typeof(TInner).IsValueType == false;
        if (innerIsRefType)
        {
            Assert.NotNull(obj.C.J);
        }

        Assert.Null(obj.C.J.A);
        Assert.Equal(7, obj.C.J.B);
    }
}

private interface ITestInterface<TNested>
{
    public string A { get; set; }
    public string B { get; set; }
    public int C { get; set; }
    public int? D { get; set; }
    public float E { get; set; }
    public float? F { get; set; }
    public bool G { get; set; }
    public bool? H { get; set; }
    public int[] I { get; set; }
    public TNested J { get; set; }
}

private class TestClass<TNested> : ITestInterface<TNested>
{
    public string A { get; set; }
    public string B { get; set; }
    public int C { get; set; }
    public int? D { get; set; }
    public float E { get; set; }
    public float? F { get; set; }
    public bool G { get; set; }
    public bool? H { get; set; }
    public int[] I { get; set; }
    public TNested J { get; set; }
}

private class TestValueType<TNested> : ITestInterface<TNested>
{
    public string A { get; set; }
    public string B { get; set; }
    public int C { get; set; }
    public int? D { get; set; }
    public float E { get; set; }
    public float? F { get; set; }
    public bool G { get; set; }
    public bool? H { get; set; }
    public int[] I { get; set; }
    public TNested J { get; set; }
}

private interface INestedInterface
{
    public string A { get; set; }
    public int B { get; set; }
}

private class NestedClass : INestedInterface
{
    public string A { get; set; }
    public int B { get; set; }
}

private struct NestedValueType : INestedInterface
{
    public string A { get; set; }
    public int B { get; set; }
}
```

I didn't look further, but there might be a way to abstract further and reduce duplicate LOC for the parameterized ctor test; up to you.

Reply:
>The first four tests in this file are very similar

I already completely refactored this for the same reason. Follow up PR is coming soon.